e-commerce/shop-backend/src/operations.ts

```diff
@@ -168,59 +239,74 @@ export class Shop {
 
   @Workflow()
   static async paymentWorkflow(ctxt: WorkflowContext, username: string, origin: string): Promise<void> {
-    const productDetails = await ctxt.invoke(Shop).getCart(username);
-    if (productDetails.length === 0) {
-      ctxt.logger.error(`Checkout for ${username} failed: empty cart`);
-      await ctxt.setEvent(checkout_url_topic, null);
-      return;
-    }
+    // Coupled with the `finally` block, this will ensure that an event is sent out of the workflow.
+    const event = new EventSender(ctxt, checkout_url_topic);
```

Comment:
Do we need this `EventSender` class?  It might be easier to do something like:

```
const event: string | null = null
try {
  ...
event = paymentSession.url
} finally {
   await ctxt.setEvent(topic, event)
}
```

I'm worried adding new utility classes is confusing.

Reply:
In general, to do that, you'd need a try/finally around the block boundary where the event must be sent, rather than the whole message guts.  In this case, that'd mean an additional try/finally around 247-271.  If that's what people want instead of a utility, we can do that.

Reply:
Got it!